### ISSUE TITLE
type alias for generic typeclass doesn't work

### ISSUE THREAD

**Issue by jcosborn** [nim-lang/Nim#4668]
The following doesn't compile on 0.14.2 or devel.

``` nimrod
type
  FooObj*[T] = object
    v*: T
  Foo1*[T] = FooObj[T]
  Foo2* = FooObj

proc foo1(x: Foo1) = echo "foo1"
proc foo2(x: Foo2) = echo "foo2"

var x: FooObj[float]
foo1(x)  # works
foo2(x)
# type mismatch: got (FooObj[system.float])
# but expected one of:
# proc foo2[Foo2](x: Foo2)
```

**Comment by Varriount** [comment]
Hm. What you probably mean is:
```
type Foo2[T] = FooObj[T]
```

I don't recall exactly what the original form does, although to my mind it should still work.

**Comment by jcosborn** [comment]
The point is to show that without the ``[T]`` it doesn't work correctly.  This should create an alias for the typeclass ``FooObj``.  Incidentally both ``x is FooObj`` and ``x is Foo1`` return true while ``x is Foo2`` gives false.

**Comment by jcosborn** [comment]
And interestingly, if I instead set ``type Foo2 = Foo1`` then it works.

**Comment by ringabout** [comment]
It works in Nim 1.2.4.

```nim
foo1
foo2
```
Fixed since 1.2.0.
https://play.nim-lang.org/#ix=2rfH
